doc/services/device_mgmt/smp_transport.rst

```diff
@@ -0,0 +1,214 @@
+.. _mcumgr_smp_transport_specification:
+
+SMP Transport Specification
+###########################
+
+The documents specifies information needed for implementing server and client
+side SMP transports.
+
+.. _mcumgr_smp_transport_ble:
+
+BLE (Bluetooth Low Energy)
+**************************
+
+SMP communication relays on a single GATT characteristic. An SMP command may
+be sent in the form of either GATT Write command or GATT Write Without Response
+command.
+SMP responses are sent in the form of a GATT Notification, specifying the same
+characteristics as GATT Write command.
+
+BLE (Bluetooth Low Energy) Characteristics
+==========================================
+
+MCUmgr Clients need to use following BLE Characteristics, when implementing
+SMP client:
+
+- **Service UUID**: `8D53DC1D-1DB7-4CD3-868B-8A527460AA84`
+- **Characteristic UUID**: `DA2E7828-FBCE-4E01-AE9E-261174997C48`
+
+All SMP communication utilizes a single GATT characteristic.  An SMP request is
+sent in the form of either 1) a GATT Write Command, or 2) a GATT Write Without
+Response command. An SMP response is sent in the form of a GATT Notification
+specifying the same characteristic that was written.
+
+The Write Without Response form is generally preferred, as an application-layer
+response is always sent in the form of a Notification. The regular Write form is
+accepted in case the client requires a GATT response to initiate pairing.
+
+If an SMP request or response is too large to fit in a single GATT command, the
+sender fragments it across several commands.  No additional framing is
+introduced when a request or response is fragmented; the payload is simply
+split among several packets.  Since Bluetooth guarantees ordered delivery of
+packets, the SMP header in the first fragment contains sufficient information
+for reassembly.
+
+Note: Zephyr currently supports only GATT Write Without Response, and sending
+responses in form of GATT Notifications.
+
+.. _mcumgr_smp_transport_uart:
+
+UART/serial and console
+***********************
+
+SMP protocol specification by MCUmgr subsystem of Zephyr uses basic framing
+of data to allow multiplexing of UART channel. Multiplexing requires
+prefixing each frame with two byte marker and terminating it with newline.
+Currently MCUmgr imposes a 128 byte limit on frame size, although there
```

Comment:
```suggestion
Currently MCUmgr imposes a 127 byte limit on frame size, although there
```

Reply:
This is wrong then, header is 3 bytes, assuming 128 byte packet, that leaves 125 bytes for data, the data is in base64, 125 / 4 = 31.25, 31.25 * 3 = 93.75 - this is not valid, therefore it allows 93 base64 encoded bytes, 93 base64 bytes is occupies 124 bytes of data, 124 + 3 = 127, so the maximum is 127, not 128.

Reply:
Unresolved